tests/test_prediction_intervals/test_base.py

```diff
@@ -0,0 +1,180 @@
+
+    intervals_pipeline = DummyPredictionIntervals(pipeline=pipeline)
+
+    intervals_pipeline.fit(ts=example_tsds)
+
+    intervals_pipeline_pred = intervals_pipeline.forecast(prediction_interval=False)
+    pipeline_pred = pipeline.forecast(prediction_interval=False)
+
+    pd.testing.assert_frame_equal(intervals_pipeline_pred.df, pipeline_pred.df)
+
+
+@pytest.mark.parametrize("pipeline_name", ("naive_pipeline", "naive_pipeline_with_transforms"))
+def test_forecast_with_fitted_pipeline(example_tsds, pipeline_name, request):
+    pipeline = request.getfixturevalue(pipeline_name)
+
+    pipeline.fit(ts=example_tsds)
+    pipeline_pred = pipeline.forecast(prediction_interval=False)
+
+    intervals_pipeline = DummyPredictionIntervals(pipeline=pipeline)
+    intervals_pipeline_pred = intervals_pipeline.forecast(prediction_interval=False)
+
+    pd.testing.assert_frame_equal(intervals_pipeline_pred.df, pipeline_pred.df)
+
+
+@pytest.mark.parametrize(
+    "expected_columns",
+    ({"target_lower", "target_upper"},),
+)
+@pytest.mark.parametrize("pipeline_name", ("naive_pipeline", "naive_pipeline_with_transforms"))
+def test_forecast_intervals_exists(example_tsds, pipeline_name, expected_columns, request):
+    pipeline = request.getfixturevalue(pipeline_name)
+
+    intervals_pipeline = DummyPredictionIntervals(pipeline=pipeline)
+    intervals_pipeline.fit(ts=example_tsds)
+
+    intervals_pipeline_pred = intervals_pipeline.forecast(prediction_interval=True)
+    columns = intervals_pipeline_pred.df.columns.get_level_values("feature")
+
+    assert len(expected_columns - set(columns)) == 0
+
+
+@pytest.mark.parametrize(
+    "expected_columns",
+    ({"target", "target_lower", "target_upper"},),
+)
+@pytest.mark.parametrize(
+    "pipeline",
+    (
+        Pipeline(model=NaiveModel(), horizon=1),
+        AutoRegressivePipeline(model=NaiveModel(), horizon=1),
+        HierarchicalPipeline(
+            model=NaiveModel(),
+            horizon=1,
+            reconciliator=BottomUpReconciliator(target_level="market", source_level="product"),
+        ),
+    ),
+)
+def test_pipelines_forecast_intervals(product_level_constant_hierarchical_ts, pipeline, expected_columns):
+    run_base_pipeline_compat_check(
+        ts=product_level_constant_hierarchical_ts, pipeline=pipeline, expected_columns=expected_columns
+    )
+
+
+@pytest.mark.parametrize(
+    "expected_columns",
+    ({"target", "target_lower", "target_upper"},),
+)
+@pytest.mark.parametrize(
+    "ensemble",
+    (
+        DirectEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=2)]),
+        VotingEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=1)]),
+        StackingEnsemble(pipelines=[get_naive_pipeline(horizon=1), get_naive_pipeline_with_transforms(horizon=1)]),
+    ),
+)
+def test_ensembles_forecast_intervals(example_tsds, ensemble, expected_columns):
```

Comment:
May be we can consider ensembles as pipelines(in fact they are) as tests are the same?